### PR TITLE
Fix typos and unused imports

### PR DESCRIPTION
**Pull request by kinow**
This is a small change with no associated Issue.

Yesterday ran the IDE linter against Cylc UI, and started doing the same with Cylc Flow. There are two issues it found that might be bugs, but they require understanding about the task pool (which is great for me learning :grimacing: probably bothering @hjoliver and others soon in Element).

This PR fixes typos, removes unused imports, and also renames two test functions that had duplicated names. They are pretty similar, but not identical, so I left both of them.

If CI passes, I think one review should be enough.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why?).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

File: cylc/flow/task_job_mgr.py [hunk 1]
```diff
@@ -719,7 +719,6 @@ def _set_retry_timers(itask, rtconfig=None, retry=True):
                 submit_delays = itask.platform['submission retry delays']
             # TODO: same for execution delays?
 
-        if retry:
```
Comment (kinow) [inline cylc/flow/task_job_mgr.py:722]: Click on the blue button up here :point_left: to see more of the code... there's another `if retry` just above this other `if retry` :+1:

**Review by hjoliver — APPROVED** [review]
Good spotting, thanks :+1:

**Review by hjoliver — APPROVED** [review]
Re-approving on latest change; I'll merge this if the tests pass.

**Comment by hjoliver** [comment]
Kicked the tests, tests/f batch 2/4 got cancelled for some reason.

File: tests/unit/test_templatevars.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def test_load_template_vars_from_string_and_file(self):
                 expected, load_template_vars(template_vars=pairs,
                                              template_vars_file=tf.name))
 
-    def test_load_template_vars_from_string_and_file(self):
+    def test_load_template_vars_from_string_and_file_2(self):
```
Comment (oliver-sanders) [inline tests/unit/test_templatevars.py:84]: Whoops.

**Comment by hjoliver** [comment]
Strangely, one test batch keeps getting cancelled on this branch :thinking:

**Comment by kinow** [comment]
> 
> 
> Strangely, one test batch keeps getting cancelled on this branch 🤔

Why are most tests being skipped like this?

>2020-12-10T06:21:24.3824085Z [11:51:24] tests/f/inheritance/01-circular.t ................................................ skipped: requires loc:local batch:background

I thought background would be the default batch handler 😕

**Comment by hjoliver** [comment]
The local and remote tests are now done in separate test batches.

**Comment by oliver-sanders** [comment]
Not sure what's going on there, surely nothing to do with this branch? Perhaps the PR?

**Comment by hjoliver** [comment]
May be fixed by #3995

**Review by oliver-sanders — APPROVED** [review]

**Comment by hjoliver** [comment]
@kinow this should pass tests after rebasing

**Comment by kinow** [comment]
One functional test job pending. :drum:

**Comment by hjoliver** [comment]
Damn it, that last batch seems to be hung :confounded:

**Comment by hjoliver** [comment]
Yep, it was cancelled on hitting the 45 minute timeout. Which suggests a hung test suite.

**Comment by hjoliver** [comment]
I downloaded the test logs and compared all listed `.t` tests with those in the repository, and none were missing. But the cancelled batch did say this at the end:
```
2020-12-14T21:08:11.2276913Z Cleaning up orphan processes
2020-12-14T21:08:11.2574787Z Terminate orphan process: pid (10082) (prove)
2020-12-14T21:08:11.2611612Z Terminate orphan process: pid (11247) (bash)
2020-12-14T21:08:11.2643628Z Terminate orphan process: pid (11730) (cylc)
2020-12-14T21:08:11.2702014Z Terminate orphan process: pid (13039) (bash)
2020-12-14T21:08:11.2733469Z Terminate orphan process: pid (13973) (cylc)
2020-12-14T21:08:11.2816499Z Terminate orphan process: pid (15408) (bash)
2020-12-14T21:08:11.2846131Z Terminate orphan process: pid (15511) (bash)
2020-12-14T21:08:11.2883995Z Terminate orphan process: pid (15859) (cylc)
2020-12-14T21:08:11.2930791Z Terminate orphan process: pid (15913) (cylc)
2020-12-14T21:08:11.3023369Z Terminate orphan process: pid (18847) (bash)
2020-12-14T21:08:11.3070552Z Terminate orphan process: pid (19850) (cylc)
```
... which seems to imply that some tests passed but left running processes behind??

**Comment by hjoliver** [comment]
However they all pass locally and don't leave any processes behind :confounded:

**Comment by hjoliver** [comment]
kicking it...

**Comment by kinow** [comment]
> kicking it...

I think I kicked it the first time the tests did not pass. I wonder if I removed one too many import somewhere? I had one unit test failing because of that, and then added back the import. Maybe the same is happening but in a functional test.

**Comment by hjoliver** [comment]
Yep, that batch is consistently hanging!

File: cylc/flow/task_job_mgr.py [hunk 3]
```diff
@@ -667,6 +666,8 @@ def _poll_task_job_callback(self, suite, itask, cmd_ctx, line):
         finally:
             log_task_job_activity(ctx, suite, itask.point, itask.tdef.name)
 
+        jp_ctx = JobPollContext(job_log_dir, **items)
```
Comment (kinow) [inline cylc/flow/task_job_mgr.py:669]: @hjoliver out of the changes in this PR, this is the only one that looks like a possible reason for the hanging workflow... maybe prior to this change it was raising an error captured somewhere else? Or maybe this is triggering some other error? The other changes are really simple, and shouldn't cause this kind of issue I guess.

Comment (hjoliver) [reply cylc/flow/task_job_mgr.py:669]: Yeah I was wondering about that too.

Comment (hjoliver) [reply cylc/flow/task_job_mgr.py:669]: Maybe that line needs to be inside the try block.

File: cylc/flow/task_job_mgr.py [hunk 4]
```diff
@@ -659,6 +659,7 @@ def _poll_task_job_callback(self, suite, itask, cmd_ctx, line):
                     (key, values[x]) for
                     x, key in enumerate(JobPollContext.CONTEXT_ATTRIBUTES))
                 job_log_dir = items.pop('job_log_dir')
+                jp_ctx = JobPollContext(job_log_dir, **items)
```
Comment (kinow) [inline cylc/flow/task_job_mgr.py:662]: Now this is created here, and in the original `try/catch` block. Let's see if the build passes after this change.

Comment (hjoliver) [reply cylc/flow/task_job_mgr.py:662]: It passed :tada:

**Review by hjoliver — APPROVED** [review]

**Pull request merged**